### ISSUE TITLE
build on raspberry pi raspbian

### ISSUE THREAD

**Issue by augre** [cryptid-org/cryptid-native#18]
Hello,

I ran into this error while building on raspbian. Do you know what is the right fix for this?

Thanks!

```
pi@raspberrypi:~/cryptid-native $ ./task.sh build
npx: installed 229 in 29.59s
Executing | gcc /home/pi/cryptid-native/src/CryptID.c /home/pi/cryptid-native/src/SignID.c /home/pi/cryptid-native/src/complex/Complex.c /home/pi/cryptid-native/src/elliptic/AffinePoint.c /home/pi/cryptid-native/src/elliptic/ComplexAffinePoint.c /home/pi/cryptid-native/src/elliptic/Divisor.c /home/pi/cryptid-native/src/elliptic/EllipticCurve.c /home/pi/cryptid-native/src/elliptic/TatePairing.c /home/pi/cryptid-native/src/identity-based/CipherTextTuple.c /home/pi/cryptid-native/src/identity-based/HashFunction.c /home/pi/cryptid-native/src/identity-based/Signature.c /home/pi/cryptid-native/src/util/RandBytes.c /home/pi/cryptid-native/src/util/Random.c /home/pi/cryptid-native/src/util/Utils.c /home/pi/cryptid-native/src/util/Validation.c /home/pi/cryptid-native/dependencies/sha/src/sha.c -I/home/pi/cryptid-native/include -I/home/pi/cryptid-native/dependencies/gmp/include -I/home/pi/cryptid-native/dependencies/sha/include -std=c99 -Wall -Wextra -Werror -c
/home/pi/cryptid-native/src/util/Utils.c: In function ‘hashToRange’:
/home/pi/cryptid-native/src/util/Utils.c:57:35: error: ‘%03X’ directive writing 3 bytes into a region of size 2 [-Werror=format-overflow=]
             sprintf((char *)hex, "%03X", h[j]);
                                   ^~~~
/home/pi/cryptid-native/src/util/Utils.c:57:13: note: ‘sprintf’ output 4 bytes into a destination of size 2
             sprintf((char *)hex, "%03X", h[j]);
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
Removing files using globs: *.o
        15 file(s) matched

```

**Comment by battila7** [comment]
Hi,

thanks for raising this issue! As stated in #19, there was indeed a bug in the library, which is going to be fixed by the pull request. I will merge it as soon as the pipeline turns green.

Bests,
Attila

**Comment by augre** [comment]
Thank you!

**Comment by battila7** [comment]
Hi,

I've just merged #19 , please let me now if it works correctly now (and feel free to close the issue in that case).

Bests,
Attila

**Comment by battila7** [comment]
Closing as potentially resolved. Please reopen, if the issue remains.
